docs/src/basics/blockchain_types.md

```diff
@@ -43,3 +43,46 @@ let my_number: b256 = 0x00000000000000000000000000000000000000000000000000000000
 let my_contract_id: ContractId = ~ContractId::from(my_number);
 let forty_two: b256 = my_contract_id.into();
 ```
+
+## `Identity` Type
+
+The `Identity` type is an enum that allows for the handling of both `Address` and `ContractId` types. This is useful in cases where either type is needed.
+
+An `Identity` is implemented as follows.
+
+```sway
+pub enum Identity {
+    Address: Address,
+    ContractId: ContractId,
+}
```

Comment:
These should all be in actual Sway files, tested in CI, and included using `{{#include }}`: https://rust-lang.github.io/mdBook/format/mdbook.html#including-files. Otherwise it's too easy for them to drift and we end up with non-working code in The Book.

Reply:
I think we need a special directory for all these little snippets like https://github.com/rust-lang/book/tree/main/listings and use anchors with them. These wouldn't be full examples like the examples directory that we have today, but small test cases that are meant only for the book but are also kept in sync with the compiler if anything changes.

Reply:
Not suggesting this to be done in this PR.. for now, maybe include a single full example with all the different cases you're showing in this section?